app/_hub/kong-inc/grpc-web/index.md

```diff
@@ -0,0 +1,158 @@
+---
+name: gRPC-Web
+publisher: Kong Inc.
+
+categories:
+  - transformations
+
+type: plugin
+
+desc: Allow browser clients to call gRPC services.
+description: |
+  A Kong plugin to allow access to a gRPC service via the gRPC-Web protocol.
+  Primarily, this means JS browser apps using the gRPC-Web library.
+  
+source_url: https://github.com/Kong/kong-plugin-grpc-web
+
+license_type: MIT
+
+kong_version_compatibility:
+  community_edition:
+    compatible:
+      - 2.0.x
+      - 1.5.x
+      - 1.4.x
+
+
+params:
+  name: grpc-web
+  route_id: true
+  protocols: ["http", "https"]
+  dbless_compatible: true
+
+  config:
+    - name: proto
+      required: false
+      default:
+      value_in_examples: path/to/hello.proto
+      description: |
+        If present, describes the gRPC types and methods.
+        Required to support payload transcoding.  When absent, the
+        web client must use application/grpw-web+proto content.
+
+---
+
+## Purpose
+
+A service that presents a gRPC API can be used by clients written in many languages, but the network specifications are oriented primarily to connections within a datacenter. [gRPC-Web] allows exposing such API to the Internet, and be consumed by brower-based JS apps.
+
+This plugin translates requests and responses between [gRPC-Web] and ["real" gRPC](https://github.com/grpc/grpc).  Supports both HTTP/1.1 and HTTP/2, over plaintext (HTTP) and TLS (HTTPS) connections.
+
+## Usage
+
+This plugin is intended to be used in a Kong Route between an HTTP endpoint and a gRPC service.
+
+Sample configuration via declarative (YAML):
+
+```yaml
+_format_version: "1.1"
+services:
+- protocol: grpc
+  host: localhost
+  port: 9000
+  routes:
+  - protocols:
+    - http
+    paths:
+    - /
+    plugins:
+    - name: grpc-web
+```
+
+Same thing via the Admin API:
+
+```bash
+$ # add the gRPC service
+$ curl -XPOST localhost:8001/services \
+  --data name=grpc \
+  --data protocol=grpc \
+  --data host=localhost \
+  --data port=9000
+
+$ # add an http route
+$ curl -XPOST localhost:8001/services/grpc/routes \
+  --data protocols=http \
+  --data name=web-service \
+  --data paths=/
+
+$ # add the plugin to the route
+$ curl -XPOST localhost:8001/routes/web-service/plugins \
+  --data name=grpc-web
+```
+
+In these examples we don't set any configuration for the plugin.  This minimal setup works for the default varieties of the [gRPC-Web protocol], which use ProtocolBuffer messages either directly in binary or with base64 encoding.  The related `Content-Type` headers are `application/grpc-web` or `application/grpc-web+proto` for binary and `application/grpc-web-text` or `application/grpc-web-text+proto`.
+
+If we wish to use JSON encoding, we have to provide the gRPC specification in a .proto file, which needs to be installed in the Kong node running the plugin.  For example:
```

Comment:
```suggestion
If we wish to use JSON encoding, we have to provide the gRPC specification in a `.proto` file, which needs to be installed in the Kong node running the plugin.  For example:
```

Reply:
Not sure how other file type extensions are shown in the docs.  Probably most are referred to by the actual acronym.  "HTML files", not ".html files", images are JPEG files, not .jpeg.  What about .js files?  Few users concern with .patch or .rst, so I guess there's not much precedence.